src/containers/EditSubjectFrontpage/components/nodes/SearchDropdown.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import styled from '@emotion/styled';
+//@ts-ignore
+import { Input, DropdownMenu } from '@ndla/forms';
+import { Spinner } from '@ndla/icons';
+import { Search } from '@ndla/icons/common';
+import { Node } from '@ndla/types-taxonomy';
+import Downshift, { GetItemPropsOptions } from 'downshift';
+import { useState } from 'react';
+import { UseQueryOptions, UseQueryResult } from '@tanstack/react-query';
+import { SearchResultBase } from '../../../../interfaces';
+import useDebounce from '../../../../util/useDebounce';
+
+const DropdownWrapper = styled.div`
+  width: 100%;
+`;
+
+const StyledSpinner = styled(Spinner)`
+  margin: 0px;
+`;
+
+interface DropdownItem<Type> {
+  id: string | number;
+  name: string;
+  image?: string;
+  description?: string;
+  originalItem: Type;
+  disabled?: boolean;
+}
+
+interface BaseParams {
+  query?: string;
+  page?: number;
+}
+
+interface Props<ParamType extends BaseParams, InnerType, ApiType, Type = ApiType> {
+  onChange: (value: InnerType) => void;
+  useQuery: (
+    params: ParamType,
+    options?: UseQueryOptions<ApiType, unknown, Type>,
+  ) => UseQueryResult<Type, unknown>;
+  params?: ParamType;
+  options?: UseQueryOptions<ApiType, unknown, Type>;
+  transform: (value: Type) => SearchResultBase<DropdownItem<InnerType>>;
+  placeholder: string;
+  preload?: boolean;
+  selectedItems?: Node[];
+  id?: string;
+  wide?: boolean;
+}
+
+const SearchDropdown = <ParamType extends BaseParams, InnerType, ApiType, Type>({
+  onChange,
+  useQuery,
+  params,
+  options = {},
+  transform,
+  placeholder,
+  preload = true,
+  selectedItems,
+  id,
+  wide = false,
+}: Props<ParamType, InnerType, ApiType, Type>) => {
+  const [query, setQuery] = useState('');
+  const [page, setPage] = useState<number>(1);
+  const debouncedQuery = useDebounce(query);
+  const allParams: ParamType = {
+    query: debouncedQuery,
+    page,
+    ...params,
+  } as ParamType;
+
+  const searchQuery = useQuery(allParams, {
+    ...options,
+    enabled: options.enabled ? options.enabled : debouncedQuery.length > 1,
+  });
+  useQuery(
+    { page: page + 1, ...allParams },
+    {
+      ...options,
+      enabled: options.enabled
+        ? options.enabled
+        : debouncedQuery.length > 1 || (!!preload && debouncedQuery.length > 1),
+    },
+  );
+  const transformed = searchQuery.data ? transform(searchQuery.data) : undefined;
+
+  const handleOnChange = (item: DropdownItem<InnerType>) => {
+    item && onChange(item.originalItem);
```

Comment:
Og pirk: Jeg foretrekker å wrape ting som ikke er en `boolean` i en conditional. Altså at dette hadde vært:

```
if (item) {
  onChange(item.originalItem)
}
```
Da reduserer man sjansen for å ende opp med ting i render-funksjonen som ikke skal være der, for eksempel!